### ISSUE TITLE
Wrong gradient calculated by `tf.autodiff.ForwardAccumulator` for API `tf.optimizers.schedules.ExponentialDecay`

### ISSUE THREAD

**Issue by eeDigitalSeal** [tensorflow/tensorflow#56836]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Current Behaviour?

```shell
Forward mode differentiation for the case below should be `0.0` but got `nan`, inconsistent with the gradient calculated in reverse mode.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
initial_learning_rate = 14.01
decay_steps = 100
decay_rate = 0.0
staircase = False
name = None

input = tf.constant(0.9369918869034664, dtype=tf.float64)

exp_decay = tf.optimizers.schedules.ExponentialDecay(initial_learning_rate, decay_steps, decay_rate,
                                                                  staircase=staircase, )

with tf.GradientTape(persistent=True) as g:
  g.watch(input)
  res_backward = exp_decay(input)
grad_backward = g.gradient(res_backward,input)
print(grad_backward)

with tf.autodiff.ForwardAccumulator(input, tf.constant(1,dtype=tf.float64)) as acc_0:
  res_forward = exp_decay(input)
grad_forward = acc_0.jvp(res_forward)
print(grad_forward)
```

### Relevant log output

```shell
tf.Tensor(0.0, shape=(), dtype=float64)
tf.Tensor(nan, shape=(), dtype=float32)
```
</details>

**Comment by tilakrayal** [comment]
@eeDigitalSeal,
I tried to execute the code with various `initial_learning_rate`, `decay_steps` and `decay_rate`. I was able to fetch the result as expected. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/be394489e88ef540e65239fa544f4c0c/untitled449.ipynb).

**Comment by eeDigitalSeal** [comment]
@tilakrayal This bug only happened when setting `decay_rate=0.0`. You can reproduce it [here](https://colab.research.google.com/drive/1SaDTNRRybR6zeyvRmq6c4ixDc68XiS-e?usp=sharing)

**Comment by tilakrayal** [comment]
@chunduriv,
I was able to reproduce the issue on tensorflow v2.8, v2.9, nightly. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/e66254f4571774cdcdb72779de96924a/56836.ipynb).

**Comment by tilakrayal** [comment]
@eeDigitalSeal,
A value too close to 1 (e.g: 0.999) might lead to the learning rate decaying very slowly, and difficult in the optimization process. and the rates extremely close to 0 (e.g: 0.0001) can cause the learning rate to decay to very small values, potentially causing numerical issues due to underflow.

A common range for decay_rate is between 0.1 and 0.9. The optimal value depends on your specific dataset and model architecture. I tried with the different **decay_rate** for `tf.optimizers.schedules.ExponentialDecay` and observed that with the `decay_rate = 0.00000000000000000000000000000000000001 ` the output was different with the another. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/c8d6d29523ce824383a6bef11ed16b45/untitled1827.ipynb). Thank you!

**Comment by svkgn4DL** [comment]
> ```shell
> ExponentialDecay
> ```

The question here is with decay_rate=0 the problem arising. Why can't it be addressed? any reason for this? may be a validation check resolves the issue.
